cmd/registry/controller/controller.go

```diff
@@ -31,6 +31,20 @@ type Action struct {
 	RequiresReceipt   bool
 }
 
+func ValidateManifest(ctx context.Context, parent string, manifest *rpc.Manifest) bool {
+	isValid := true
+	for _, resource := range manifest.GeneratedResources {
+		log.FromContext(ctx).Debugf("Validating entry: %v", resource)
+		if errs := validateGeneratedResourceEntry(parent, resource); errs != nil {
+			for _, err := range errs {
+				log.FromContext(ctx).WithError(err).Errorf("Invalid manifest entry")
```

Comment:
>The original thinking behind this was, because this validate function is used outside of the package (in upload manifest and upload artifact commands) where the general pattern is usually returning a single error and a response. That error is then just logged as is without any processing.

IMO we should give those commands control over their log output. If we return multiple errors they can choose to loop and log each one, replace the detailed errors with a summary, ignore the errors, etc. It gives them flexibility.

>Another reason to log errors here is so that we can specifically mention which errors belong to which manifest entry, with the log entry Validating entry %v. If we pass a list of errors to the calling function, it will have to traverse through the whole list and log a combined list of errors without identifying which entry it belongs to (which we can solve by generating more informative errors)

Anything we can log in this function can be returned as an error with the same information. We can give access to any level of detail that callers might want, e.g. validation for the entire manifest, or specific parts.

>but the purpose of this function is to just return a yes/no answer to the callee.

Why do callers need that? It looks like returning false will just result in an error being propagated back to the root command, leading to an error with very little detail about the true cause. This seems like another reason to return errors instead.

Reply:
Done. Now `manifest upload` and `artifact upload` commands are logging the errors